File: typescript/vercel-ai-sdk/zero-bloat-jq-filter/test_tools.ts
import { Client } from '@modelcontextprotocol/sdk/client/index.js';
import { StreamableHTTPClientTransport } from '@modelcontextprotocol/sdk/client/streamableHttp.js';
import * as dotenv from 'dotenv';
dotenv.config();

async function main() {
    const neoniaApiKey = process.env.NEONIA_API_KEY;
    const headers = neoniaApiKey ? { "Authorization": `Bearer ${neoniaApiKey}` } : undefined;
    const transport = new StreamableHTTPClientTransport(new URL("https://mcp.neonia.io/mcp?tools=neo_data_jq_filter"), {
        requestInit: headers ? { headers } : undefined
    });
    const client = new Client({ name: "test", version: "1.0.0" }, { capabilities: {} });
    await client.connect(transport);
    const toolsList = await client.listTools();
    console.log(JSON.stringify(toolsList.tools, null, 2));
    transport.close();
}
main();

File: test_tools.py
import asyncio, os, httpx
from mcp.client.streamable_http import streamable_http_client
from mcp import ClientSession
async def main():
    url = "https://mcp.neonia.io/mcp?tools=neo_web_json_fetch,neo_data_jq_filter"
    headers = {"Authorization": f"Bearer {os.getenv('NEONIA_API_KEY')}"} if os.getenv('NEONIA_API_KEY') else {}
    async with streamable_http_client(url, http_client=httpx.AsyncClient(headers=headers, timeout=60.0)) as streams:
        async with ClientSession(streams[0], streams[1]) as session:
            await session.initialize()
            mcp_tools_response = await session.list_tools()
            print("Tools available:")
            for t in mcp_tools_response.tools:
                print(f"- {t.name}")
asyncio.run(main())
